js/infobar.js

```diff
@@ -0,0 +1,478 @@
+$(function() {
+
+    'use strict';
+
+    var config = {
+        // The $infoBar HTML element
+        $infoBar: {},
+        $message: {},
+        $acceptButton: {},
+        $cancelButton: {},
+        // info bar(s) the user wish to use
+        bars: [],
+        // The current instant if an InfoBar
+        bar: {},
+        opened:false,
+        json: {},
+        lang: '',
+        // list of available languags from either a list of hreflang links
+        // or options from a select element. Populated by @getAvailableLangs
+        availableLangs: {},
+        // Array of one or more user languages as exposed by the user agent
+        acceptLangs:[]
+    };
+
+    var InfoBar = function (id, name) {
+        this.id = id;
+        this.name = name;
+        this.disabled = false;
+        this.prefName = 'infobar.' + id + '.disabled';
+
+        // Read the preference
+        try {
+            if (sessionStorage.getItem(this.prefName) === 'true') {
+                this.disabled = true;
+            }
+        } catch (ex) {
+            console.log('Error thrown while getting disabled state from sessionStorage: ', ex);
+        }
+
+        // If there is already another $infoBar, don't show this
+        if (config.$infoBar.filter(':visible').length) {
+            this.disabled = true;
+        }
+    };
+
+    /**
+     * Returns selected language's BiDi layout.
+     * False = left-to-right layout
+     * True = right-to-left layout
+     * @param {sting} lang - A language string such as en-US
+     */
+    InfoBar.prototype.getLanguageBidi = function(lang) {
+        // Languages using BiDi (right-to-left) layout
+        var LANGUAGES_BIDI = ['he', 'ar', 'fa', 'ur'];
+
+        if (!lang) {
+            return false;
+        }
+
+        var baseLang = lang.split('-')[0];
+        return $.inArray(baseLang, LANGUAGES_BIDI) > -1;
+    };
+
+    /**
+     * Normalize the user language in the form of ab or ab-CD
+     * @param {string} lang - The language string to be normalized
+     */
+    InfoBar.prototype.normalize = function (lang) {
+     return lang.replace(/^(\w+)(?:-(\w+))?$/, function (m, p1, p2) {
+         return p1.toLowerCase() + ((p2) ? '-' + p2.toUpperCase() : '');
+     });
+    };
```

Comment:
Small nit - indentation above is inconsistent. (Sorry, this is my own personal pet peeve. :grimacing:)